### PR TITLE
Speedups

### PR DESCRIPTION
**Pull request by ajwheeler**
This pull request makes two changes informed by profiling.
1. Tweaks to how the arithmetic for `voigt` is implemented.  I'm curious if you would change anything else
2. Use of a static array defined outside of `_hydrogenic_bf_cross_section`.  I think your initial version would have been nearly as fast as this, sorry for requesting the regression.

#32 should be merged before this.  I will do the force push thing to reset the github diff if necessary.

### CODE REVIEW

**Review by mabruzzo — APPROVED** [review]
Looks good to me. I had a minor question a minor suggestion for cleaner code.

I also described one last potential optimization idea for `voigt`. But that idea's a little extreme. Don't feel obliged to look into it

File: src/continuum_opacity/hydrogenic_bf_ff.jl [hunk 1]
```diff
@@ -35,6 +36,14 @@ _HztoeV(freq) = freq*hplanck_eV
 # In the following equation, e is the elementary charge in units of statcoulombs
 const _bf_σ_const = 2.815e29 # = 64*π⁴e^10 mₑ / (c h⁶ 3√3)
 
+const _bf_σ_coef = SA[(0.9916,  2.719e13, -2.268e30),
```
[the remark this answers is not in the document]
Comment (ajwheeler) [reply src/continuum_opacity/hydrogenic_bf_ff.jl:39]: That's correct.

File: src/line_opacity.jl [hunk 2]
```diff
@@ -174,8 +185,11 @@ function voigt(α, v)
             ψ = 0.979895023 - 0.962846325α + 0.532770573α^2 - 0.122727278α^3
             ψ * (M₀ + M₁*α + M₂*α^2 + M₃*α^3 + M₄*α^4)
         else #α > 1.4 or (α > 0.2 and α + v > 3.2)
-            u = sqrt(2) * (α^2 + v^2)
-            sqrt(2/π) * (α/u) * (1 + (3v^2 - α^2)/u^2 + (15v^4 - 30α^2*v^2 + 2α^4)/u^4)
+            α2 = α*α
+            v2 = v*v
+            invu = 1/sqrt(2) / (α2 + v2)
+            invu2 = invu*invu
+            sqrt(2/π) * (α*invu) * (1 + (3v2 - α2 + (15v2*v2 - (15*v2 + α2)*2α2)*invu2)*invu2)
```
[the remark this answers is not in the document]
Comment (ajwheeler) [reply src/line_opacity.jl:187]: Is the idea that `voigt` would compute `v = r * α` at the top?  `v` is  needed to choose between branches "3", and "4", so I think we need it every time. Or would you suggest `α*(1 + r) < 3.2`?

Does it make sense to define `voigt(α, v, r)`?

**Comment by ajwheeler** [comment]
I pushed the constant-time `move_bounds`, which makes the function invisible in the flame graph.  I can't tell you exactly how much faster it is because I'm doing more stuff on my computer right now and it's not a fair comparison, but I think it should be 10-20% for many problems.

File: src/line_opacity.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajwheeler) [reply src/line_opacity.jl:200]: I was messing around with this and it is a bit faster, but there's also a numerical stability problem near the core of Lorentz-dominated lines.

It reminded me, though, that before we invest tons of time micro-optimizing this, there is a bug in the voigt code.  In fact, it's in the branch we have been working on!
image

This is noted in  #1, but I hadn't thought about it in a while.  I'm going to try to figure it out...

[the remark this answers is not in the document]
Comment (ajwheeler) [reply src/line_opacity.jl]: I no longer think this is a bug, just a limit of the approximation.  Will post an issue documenting the situation soon.

**Comment by ajwheeler** [comment]
Interestingly, passing in `r` slows things down considerably, at least as I had handled it.  This latest commit is a bit faster, though.  It gets below 6.5s on the test we have been using most of the time.

File: src/line_opacity.jl [hunk 3]
```diff
@@ -92,6 +92,21 @@ function scaled_vdW(vdW::Tuple{F, F}, m, T) where F <: AbstractFloat
...
+end
+function _dont_oob(n, l)
+    if n < 1
+        1
+    elseif n > l
+        l
+    else
+        n
+    end
+end
```
Comment (mabruzzo) [inline src/line_opacity.jl:109]: I have a minor suggestion: replace `_dont_oob` with the [`clamp`](https://docs.julialang.org/en/v1/base/math/#Base.Math.clamp) function from the standard library

```suggestion
function move_bounds(λs::AbstractRange, lb, ub, λ₀, window_size)
    len = length(λs)
    lb = clamp(Int(cld(λ₀ - window_size - λs[1], step(λs)) + 1), 1, len)
    ub = clamp(Int(fld(λ₀ + window_size - λs[1], step(λs)) + 1), 1, len)
    lb,ub
end
```

**Review by mabruzzo — APPROVED** [review]
Looks great to me! I have one minor suggestion for your new `move_bounds` function to make things slightly more concise.

But, feel free to ignore it.

**Pull request merged**